### ISSUE TITLE
Use .MAP file instead of .PDB

### ISSUE THREAD

**Issue by squidcc** [diasurgical/devilution-comparer#11]
Currently we have the awkward situation where the VC6 linker is used to generate builds with a PDB file, since the VC5 linker can't process PDBs produced by the VC6 compiler. This means the builds end up with different layout, different startup stub, different static constructors etc... 
But the VC5 linker can produce a .map file which includes the locations of all functions and variables in the final build, just like the PDB information. Could we switch to using it instead so we don't have to make separate builds?

**Comment by seritools** [comment]
Sounds more than reasonable! I'll try to look into it next weekend.

**Comment by seritools** [comment]
So the .map file contains mangled names - does someone know what would be the easiest way to unmangle them? VC6 and up have a `UNDNAME.EXE` which can do that, but cannot be supplied like `cvdump.exe` I think (because of licensing I'd say)

**Comment by mewmew** [comment]
Perhaps https://pypi.org/project/cxxfilt/ or https://github.com/benfred/py-cpp-demangle could be used for demangling of symbol names?

**Comment by seritools** [comment]
Looked into it, it basically just uses "__cxa_demangle", which most probably doesn't work with the MSVC mangling scheme (that's different from other compilers as far as I've read). It'd also make the tool not work under windows.

Windows itself has [UnDecorateSymbolName](https://docs.microsoft.com/en-us/windows/desktop/api/Dbghelp/nf-dbghelp-undecoratesymbolname) as alternative (with the reverse problem of not working under other OSes of course)

**Comment by AJenbo** [comment]
https://github.com/nico/demumble

**Comment by AJenbo** [comment]
This one might also be interested, it seams more a drop in replacement for cvdump.exe (same license, prebuild Windows exe)
https://github.com/mooware/msvcfilt

**Comment by seritools** [comment]
Yeah, the second one probably is the simplest one for this - both have to be edited to output just the function names (without signatures) though.
